### PR TITLE
Set type as private for onboarding board

### LINKED ISSUE #2661: Bug: Onboarding board is not created as a private board
The onboarding board should be a private board. Currently, it's being created as a public board.

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #2661

### CODE REVIEW

**Review by sbishel — CHANGES_REQUESTED** [review]

File: server/app/onboarding.go [hunk 1]
```diff
@@ -81,5 +81,17 @@ func (a *App) createWelcomeBoard(userID, teamID string) (string, error) {
...
+	newType := model.BoardTypePrivate
+
+	patch := &model.BoardPatch{
+		Type: &newType,
+	}
+
+	if _, err := a.PatchBoard(patch, bab.Boards[0].ID, userID); err != nil {
+		return "", err
+	}
+
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/app/onboarding.go:95]: We already have `asTemplate` modifier in `DuplicateBoard`, I don't want to add another `asType` modifier in the params.

I think we can create a ticket to handle this in the next release. I'm thinking of replacing the `toTeam` and `asTemplate` params of `duplicateBoard` with `model.BoardPatch`. That we can perform any level of modification while duplicating. For example, updating the title to add a ` (copy)` suffix to duplicated board. Would that be okay for now?

[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/app/onboarding.go:95]: Created item for today's discussion as well - https://community-daily.mattermost.com/boards/workspace/b1ni7nprjpfwpb566k1gk8z9nw/886d2280-e88e-4065-8c85-1b62578f192e/f30dcb19-a0e8-45d2-8768-a9489f6e7fe8/cndz3kohx3by4pn74y6jci69jko

**Review by jespino — APPROVED** [review]
LGTM

**Review by sbishel — APPROVED** [review]

**Pull request merged**